lib/wallet/walletdb.js

```diff
@@ -206,6 +206,65 @@ class WalletDB extends EventEmitter {
       wallet.id, wallet.wid, addr.toString(this.network));
 
     this.primary = wallet;
+
+    await this.migrateChange();
+  }
+
+  /**
+   * Run change address migration.
+   * @param {Wallet} wallet
+   */
+
+  async migrateChange() {
+    if (this.options.memory)
+      return;
+
+    if (await this.db.has(layout.M.encode(0)))
+      return;
+
+    const {prefix} = this.options;
+    const bak = path.join(prefix, `wallet.${Date.now()}.bak`);
+    const b = this.db.batch();
+
+    b.put(layout.M.encode(0), null);
+
+    if (!this.state.marked) {
+      await b.write();
+      return;
+    }
+
+    this.logger.warning('Running change address migration.');
+    this.logger.warning('Backing up database to %s.', bak);
+
+    await this.db.backup(bak);
+
+    const wids = await this.db.keys({
+      gte: layout.W.min(),
+      lte: layout.W.max(),
+      parse: key => layout.W.decode(key)[0]
+    });
+
+    let total = 0;
+
+    for (const wid of wids) {
+      const wallet = await this.get(wid);
+
+      this.logger.info('Regenerating change addresses (id=%s, wid=%d).',
+                       wallet.id, wid);
+
+      total += await wallet.migrateChange(b);
+    }
+
+    await b.write();
+
+    if (total > 0) {
+      this.logger.warning('Regenerated %d change addresses.', total);
+      this.logger.warning('Rescanning...');
+
+      await this.scan();
```

Comment:
ooh, what if the user is running pruned? (probably not very likely at this point, but makes me think this maybe should not be automatic)